README.md

```diff
@@ -1,28 +1,29 @@
-# 🔌 create-eth Extensions
+# ERC-20 Token Extension for Scaffold-ETH 2
 
-This repository holds all the BG curated extensions for [create-eth](https://github.com/scaffold-eth/create-eth), so you can extend the functionality of your Scaffold-ETH project.
+This extension introduces an ERC-20 token contract and demonstrates how to use interact with it, including getting a holder balance and transferring tokens.
 
-## Usage
+The ERC-20 Token Standard introduces a standard for Fungible Tokens ([EIP-20](https://eips.ethereum.org/EIPS/eip-20)), in other words, each Token is exactly the same (in type and value) as any other Token.
+
+The ERC-20 token contract is implemented using the [ERC-20 token implementation](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol) from OpenZeppelin.
+
+## Installation
 
 You can install any of the extensions in this repository by running the following command:
 
 ```bash
-npx create-eth@latest -e <extension-name>
+npx create-eth@latest -e erc-20
 ```
 
-## Available Extensions
+## 🚀 Setup extension
 
-- [subgraph](https://github.com/scaffold-eth/create-eth-extensions/tree/subgraph): This Scaffold-ETH 2 extension helps you build and test subgraphs locally for your contracts. It also enables interaction with the front-end and facilitates easy deployment to Subgraph Studio.
-- [eip-712](https://github.com/scaffold-eth/create-eth-extensions/tree/eip-712): An implementation of EIP-712, allowing you to send, sign, and verify typed messages in a user-friendly manner.
-- [ponder](https://github.com/scaffold-eth/create-eth-extensions/tree/ponder): This Scaffold-ETH 2 extension comes pre-configured with [ponder.sh](https://ponder.sh), providing an example to help you get started quickly.
-- [onchainkit](https://github.com/scaffold-eth/create-eth-extensions/tree/onchainkit): This Scaffold-ETH 2 extension comes pre-configured with [onchainkit](https://onchainkit.xyz/), providing an example to help you get started quickly.
+**Hardhat:** set your burner wallet address in the deploy script at `packages/hardhat/deploy/01_deploy_se2_token.ts`, line 25 (const frontendAddress).
 
-## Create your own extension
+**Foundry:** set your burner wallet address in the deploy script at `packages/foundry/script/Deploy.s.sol`, line 36 (address frontendAddress).
 
-You can extend Scaffold-ETH by creating your own extension. To do so, you need to create a new repository with the following structure:
+Deploy your contract running `yarn deploy`
```

Comment:
Maybe we can think about a different workflow?

I don't like the contract not deploying out of the box, and having to manually enter the address. Maybe we do a (free or not) mint option?

Let's see what others think too!

Reply:
Added the free mint and removed entering the address on deployment.